exonum/src/runtime/mod.rs

```diff
@@ -12,6 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Transactions runtime.
+//!
+//! The module containing common building blocks for creating runtimes for the Exonum blockchain.
+//!
+//! Each runtime contains specific services to execute transactions, process events,
+//! and provide user APIs, e.t.c. There is a unified dispatcher that redirects all calls
+//! and requests to the appropriate runtime environment. Thus, users work with it, and not
+//! with a specific runtimes.
+//!
+//! # Service life cycle
+//!
+//! 1. Each runtime has own [artifacts] registry from which users can deploy them. The artifact
+//! identifier is required by the runtime for constructing service instances. In other words,
+//! an artifact identifier means same as class name, and a specific service instance is
+//! the class instance.
+//!
+//! 2. Each validator should request the dispatcher to deploy an artifact and send confirmation
+//! if this request is successful. Then, if the number of confirmations is equal to the total
+//! number of validators, each validator calls the dispatcher to register the artifact as deployed.
+//! After that validators can send requests to start new services instances from this artifact.
+//!
+//! 3. To start a new service instance, each validator should send request to dispatcher.
+//! Each request contains the artifact identifier, instance name, and instance configuration parameters.
+//! Then, as in the previous case, if the number of confirmations is equal to the total number of validators,
+//! each validator calls dispatcher to start a new service instance.
+//!
+//! 4. // TODO modify instance configuration procedure.
+//!
+//! 5. // TODO stop instance procedure.
+//!
+//! Each `Exonum` transaction is verified instance of [`AnyTx`] message.
+//!
+//! # Transaction life cycle
+//!
+//! 1. An Exonum client creates a transaction message, including [`CallInfo`] information to
+//! find the corresponding handler to execute, serialized transaction parameters as a payload,
+//! and signs the message with the author's key pair.
+//!
+//! 2. The client transmits the message to one of the Exonum nodes in the network.
+//! The transaction is identified by the hash of the corresponding message.
+//!
+//! 3. Node verifies that the transaction has been correctly signed.
```

Comment:
I'd propose like this ... @Maria-Nosyk help 🙂
```suggestion
//! 3. Node verifies that transaction for a correctness of the signature.
```